lib/synapse/haproxy.rb

```diff
@@ -742,6 +742,14 @@ def generate_backend_stanza(watcher, config)
           b = "\tserver #{backend_name} #{backend['host']}:#{backend['port']}"
           b = "#{b} cookie #{backend_name}" unless config.include?('mode tcp')
           b = "#{b} #{watcher.haproxy['server_options']}" if watcher.haproxy['server_options']
+          if backend.has_key?('weight')
+            # Check if server_options already contains weight, is so log a warning
+            if watcher.haproxy['server_options'].include? 'weight'
```

Comment:
@jolynch, that's a good point. To reduce the number of iteration for this PR, could you suggest what message you'd like to see when haproxy_server_options also includes weight?